File: Studio Classico/src/main/java/es/uco/iw/model/Report.java
package es.uco.iw.model;

public class Report {
	private int id;
	private User user;
	private Piece piece;
	private String description;
	
	public Report(int id, User user, Piece piece, String description){
		setID(id);
		setUser(user);
		setPiece(piece);
		setDescription(description);
	}
	
	public int getID() {
		return id;
	}
	
	public User getUser() {
		return user;
	}
	
	public Piece getPiece() {
		return piece;
	}
	
	public String getDescription() {
		return description;
	}
	
	public void setID(int id) {
		this.id = id;
	}
	
	public void setUser(User user) {
		this.user = user;
	}
	
	public void setPiece(Piece piece) {
		this.piece = piece;
	}
	
	public void setDescription(String description) {
		this.description = description;
	}
}

File: Studio-Classico/app/Models/Report.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Report extends Model
{
    use HasFactory;

	protected $table = "reports";

	protected $fillable = [
		'description'
	];

	public function piece(){
		return $this -> belongsTo(Piece::class);
	}

	public function user(){
		return $this -> belongsTo(User::class);
	}
    /*private $id;
	private $user;
	private $piece;
	private $description;
	
	function __construct($id, $user, $piece, $description){
		$this -> setID($id);
		$this -> setUser($user);
		$this -> setPiece($piece);
		$this -> setDescription($description);
	}
	
	function getID() {
		return $this -> id;
	}
	
	function getUser() {
		return $this -> user;
	}
	
	function getPiece() {
		return $this -> piece;
	}
	
	function getDescription() {
		return $this -> description;
	}
	
	function setID($id) {
		$this -> id = $id;
	}
	
	function setUser($user) {
		$this -> user = $user;
	}
	
	function setPiece($piece) {
		$this -> piece = $piece;
	}
	
	function setDescription($description) {
		$this -> description = $description;
	}*/
}
